Get and set listeners

elihart opened the pull request:
Allow request listeners to be accessed and changed after the request is constructed. This allows the request manager to intercept the request and replace the initial listener.

This also changes "markDelivery" to specify the type of delivery - cache, network, or error - so we can know where the delivered response is coming from.

elihart committed:
Make request listeners accessable and settable

elihart committed:
Differentiate delivery types

nwadams commented:
lgtm

carolineleung commented on library/src/main/java/com/android/volley/Request.java:
```diff
@@ -579,8 +586,8 @@ public RetryPolicy getRetryPolicy() {
      */
-    public void markDelivered() {
-        mResponseDelivered = true;
+    public void markDelivery(DeliveryType type) {
```
This should be `setDeliveryType` instead

carolineleung commented on library/src/main/java/com/android/volley/Request.java:
```diff
@@ -579,8 +586,8 @@ public RetryPolicy getRetryPolicy() {
      * Mark this request as having a response delivered on it.  This can be used
```
This javadoc is also no longer valid for the interface, needs updated

carolineleung commented:
Discussed in person re my concerns, :+1: after enhancing the javadocs

elihart committed:
Remove error delivery type and add javadoc

elihart commented:
Added javadoc - let me know if it's not clear.

I also removed the Error response type. I realized that by setting that we remove a record of having a real response delivered, and it would've allowed an error to be returned after a cache response. The Error type wasn't ever used, I just thought it would be useful in the future. We don't need it.

elihart committed:
Typo

lxdvs commented:
:shipit:

elihart committed:
Javadoc
